projects/exui-common-lib/src/lib/services/idle/idle.services.ts

```diff
@@ -0,0 +1,80 @@
+import { Injectable } from '@angular/core';
+import {Idle, DocumentInterruptSource } from '@ng-idle/core';
+import {select, Store} from '@ngrx/store';
+import * as fromRoot from '../../store';
+import {
+  delay,
+  distinctUntilChanged,
+  map
+} from 'rxjs/operators';
+import {Keepalive} from '@ng-idle/keepalive';
+import {IdleConfigModel} from '../../models/idle-config.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class IdleService {
+  constructor(
+    private idle: Idle,
+    private keepalive: Keepalive,
+    private store: Store<fromRoot.State>
+  ) {}
+
+  public init(idleConfig: IdleConfigModel): void {
+    this.idle.setIdleName(idleConfig.idleServiceName);
+    this.idle.setTimeout(idleConfig.timeout);
+
+    const interrupt =
+      new DocumentInterruptSource('mousedown keydown DOMMouseScroll mousewheel touchstart touchmove scroll');
+    this.idle.setInterrupts([interrupt]);
+
+    // adding delay so that user can click on sign out before the modal shuts
+    this.idle.onIdleEnd.pipe(delay(250)).subscribe(() => {
+      console.log('No longer idle.');
+      this.dispatchModal(undefined, false);
+    });
+
+    this.idle.onTimeout.subscribe(() => {
+      console.log('Timed out!');
+      this.dispatchSignedOut();
+      this.dispatchModal(undefined, false);
+    });
+
+    this.idle.onIdleStart.subscribe(() => {
+      console.log('You\'ve gone idle!');
+    });
+
+    this.idle.onTimeoutWarning.pipe(
+      map(sec => (sec > 60) ? Math.ceil(sec / 60) + ' minutes' : sec + ' seconds'),
+      distinctUntilChanged()
+    ).subscribe((countdown) => {
+      this.dispatchModal(countdown, true);
+    });
+
+    this.keepalive.interval(idleConfig.keepAliveInSeconds);
+    this.keepalive.onPing.pipe(delay(250)).subscribe(() => {
+      console.log('Keep alive');
```

Comment:
console logs should be removed

Reply:
i need this for debugging sometimes difficult to know when u implement this component what is going on ulness you have some debugger

Reply:
as above